lib/stripeHelpers.js

```diff
@@ -1,37 +1,26 @@
 const Stripe = require('stripe')
 const stripe = Stripe(process.env.STRIPE_SECRET_KEY)
 
-const findOrCreateCustomer = async ({ email, meta, source }) => {
-  const existingCustomerRes = await stripe.customers.list({ email })
-  if (existingCustomerRes.data.length) {
-    const customer = existingCustomerRes.data[0]
-    if (customer && customer.sources && customer.sources.total_count > 0) {
+const findOrCreateCustomer = async ({ email, name, source, meta }) => {
+  const { data } = await stripe.customers.list({ email })
+  if (data.length) {
+    const customer = data[0]
+    if (customer && customer?.sources?.total_count > 0 && source) {
```

Comment:
within the if (data.length) on line 6, we should add 

```
if (data.length && source)
```
since we never want to update a customer if they dont have a new source

then you can remove the && source from line 8 where it currently is